### PR TITLE
[ios] Update Quick and Nimble dependencies

### PR DESCRIPTION
**Pull request by wschurman**
# Why

One feature I'm looking to use in nimble is assertion expectations: https://github.com/Quick/Nimble#swift-assertions

Support for these running on AMD was added in a more recent version.

# How

Upgrade deps.

# Test Plan

Wait for full test run.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by tsapeta** [comment]
Afaik there were some breaking changes in one of the previous Quick versions, so this sounds like a bigger task 😉

**Comment by wschurman** [comment]
Expo Go build is failing from base PR. Will fix there. This is ready to review though since it only touches unit tests.

File: packages/expo-dev-launcher/ios/Tests/EXDevLauncherPendingDeepLinkRegistryTest.swift [hunk 1]
```diff
@@ -44,7 +44,7 @@ class EXDevLauncherPendingDeepLinkRegistryTest: QuickSpec {
 
       expect(registry.pendingDeepLink).to(beNil())
       expect(listener.lastDeepLink?.absoluteString).to(equal("http://localhost:1234"))
-      expect(listener.lastDeepLink).to(be(consumedURL))
+      expect((listener.lastDeepLink! as NSURL) === (consumedURL! as NSURL)).to(beTrue())
```
Comment (ide) [inline packages/expo-dev-launcher/ios/Tests/EXDevLauncherPendingDeepLinkRegistryTest.swift:47]: Why this change? Having the assertion library do the equality check produces a clearer error message

File: packages/expo-modules-core/ios/Tests/ExceptionsSpec.swift [hunk 2]
```diff
@@ -73,7 +73,7 @@ final class ExceptionsSpec: ExpoSpec {
       let b = TestException().causedBy(a)
       let c = TestException().causedBy(b)
 
-      expect(c.rootCause) === a
+      expect((c.rootCause as! TestException) === a).to(beTrue())
```
Comment (ide) [inline packages/expo-modules-core/ios/Tests/ExceptionsSpec.swift:76]:
```suggestion
      expect((c.rootCause as! TestException)).to(beEqual(a))
```
Something like this possible? (Suggestion is not real code)

File: packages/expo-modules-core/ios/Tests/ExpoModulesSpec.swift [hunk 3]
```diff
@@ -61,7 +67,7 @@ class ExpoModulesSpec: ExpoSpec {
         let dict = try runtime.eval("expo.modules.TestModule").asDict()
 
         dict.forEach { (key: String, value: Any) in
-          expect(value) === dict[key]!
+          expect((value as! NSObject) === dict[key]! as! NSObject).to(beTrue())
```
Comment (ide) [inline packages/expo-modules-core/ios/Tests/ExpoModulesSpec.swift:70]: Same comment about having the assertion library do the equality check

File: packages/expo-modules-core/ios/Tests/FunctionSpec.swift [hunk 4]
```diff
@@ -103,7 +103,8 @@ class FunctionSpec: ExpoSpec {
 
               expect(value).notTo(beNil())
               expect(value).to(beAKindOf(String.self))
-              expect(value).to(be(dict["property"]))
+//              expect(value).to(equal(dict["property"]))
```
Comment (ide) [inline packages/expo-modules-core/ios/Tests/FunctionSpec.swift:106]: Mean to revisit this?

File: packages/expo-updates/ios/Tests/UpdateSpec.swift [hunk 5]
```diff
@@ -7,13 +7,13 @@ import ExpoModulesTestCore
...
-    UpdatesConfig.EXUpdatesConfigRuntimeVersionKey: "1",
-    UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://exp.host/@test/test"
-  ])
-  let database = UpdatesDatabase()
-  
-  override func spec() {
+  override class func spec() {
+    let config = try! UpdatesConfig.config(fromDictionary: [
+      UpdatesConfig.EXUpdatesConfigRuntimeVersionKey: "1",
+      UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://exp.host/@test/test"
```
Comment (ide) [inline packages/expo-updates/ios/Tests/UpdateSpec.swift:13]:
```suggestion
      UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://u.expo.dev/00000000-0000-0000-0000-000000000000"
```

Minor but can replace classic url with EAS

File: packages/expo-updates/ios/Tests/NewUpdateSpec.swift [hunk 6]
```diff
@@ -7,13 +7,13 @@ import ExpoModulesTestCore
...
 
 class NewUpdateSpec : ExpoSpec {
-  let config = try! UpdatesConfig.config(fromDictionary: [
-    UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://exp.host/@test/test",
-    UpdatesConfig.EXUpdatesConfigRuntimeVersionKey: "1",
-  ])
-  let database = UpdatesDatabase()
+  override class func spec() {
+    let config = try! UpdatesConfig.config(fromDictionary: [
+      UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://exp.host/@test/test",
```
Comment (ide) [inline packages/expo-updates/ios/Tests/NewUpdateSpec.swift:12]:
```suggestion
      UpdatesConfig.EXUpdatesConfigUpdateUrlKey: "https://u.expo.dev/00000000-0000-0000-0000-000000000000",
```

File: packages/expo-updates/ios/Tests/LegacyUpdateSpec.swift [hunk 7]
```diff
@@ -7,17 +7,17 @@ import ExpoModulesTestCore
 import EXManifests
 
 class LegacyUpdateSpec : ExpoSpec {
```
Comment (ide) [inline packages/expo-updates/ios/Tests/LegacyUpdateSpec.swift:9]: Unrelated to this PR but do we need LegacyUpdate tests? Holding on to them till SDK 51's Expo Go drops 49?

**Review by ide — APPROVED** [review]
Approving but left some smaller comments.

**Comment by tsapeta** [comment]
> Expo Go build is failing from base PR. Will fix there. This is ready to review though since it only touches unit tests.

#25467 should fix this issue

File: packages/expo-modules-core/ios/Tests/ExpoModulesSpec.swift [hunk 8]
```diff
@@ -4,8 +4,14 @@ import ExpoModulesTestCore
 
 @testable import ExpoModulesCore
 
+public extension Equatable {
+  static func equate(_ any0: some Any, _ any1: some Any) -> Bool {
```
[the remark this answers is not in the document]
Comment (wschurman) [reply packages/expo-modules-core/ios/Tests/ExpoModulesSpec.swift]: Weird, seems like this was auto-added? I'll remove

File: packages/expo-modules-core/ios/Tests/ExpoModulesSpec.swift (hunk 3 above)
Comment (wschurman) [reply packages/expo-modules-core/ios/Tests/ExpoModulesSpec.swift]: done

File: packages/expo-modules-core/ios/Tests/FunctionSpec.swift [hunk 9]
```diff
@@ -103,7 +103,8 @@ class FunctionSpec: ExpoSpec {
...
+              expect(value)
```
[the remark this answers is not in the document]
Comment (wschurman) [reply packages/expo-modules-core/ios/Tests/FunctionSpec.swift]: Yeah, sorry, this was a work-in-progress that I missed when making this ready for review. fixed

File: packages/expo-updates/ios/Tests/LegacyUpdateSpec.swift (hunk 7 above)
Comment (wschurman) [reply packages/expo-updates/ios/Tests/LegacyUpdateSpec.swift:9]: Yep.

**Review by douglowder — APPROVED** [review]

File: apps/native-tests/ios/NativeTests.xcodeproj/project.pbxproj
Comment (wschurman) [inline apps/native-tests/ios/NativeTests.xcodeproj/project.pbxproj:1]: going to revert this file. no idea why it was generated this way but ld_classic seems wrong. will re-run `pod install` in a separate PR.

**Comment by wschurman** [comment]
Cool, addressed comments and unit tests passed. E2E test failures look unrelated. Test suite timed out. Going to land after addressing self-comment to hopefully fix a few of the other failing tests suites.
